### DISCUSSION TITLE
Why is the plugin manifest not bundled into the binary?

### DISCUSSION THREAD

**Discussion by multivac61** [DISTRHO/Cardinal#769] [Q&A]
### Description

Thank you for all your hard work on this project. Very impressive.

Please correct me if I am wrong, but wouldn't it be easier to bundle the plugins manifest inside the binary and skip the installation phase, e.g., to "/Library/Application Support/Cardinal/"? I see this as a huge improvement over VCV Rack, where the manifest is not known at runtime due to networking.

Packaging the manifest into the binary would also make it easier to package and distribute (just copy/paste the binary)

BTW I have packaged couple new module vendors in my [fork](https://github.com/DISTRHO/Cardinal/compare/main...multivac61:Cardinal:main). Happy to provide individual PRs if you want.

**Comment by multivac61** [comment]
Additional benefit is that you could have multiple versions of Cardinal installed at the same time without having them all depend on the same manifest.

**Comment by falkTX** [comment]
It already does that on other OSes, just not on macOS.
The main reason is binary size. on macOS we are already 100+Mb per binary, which then we have JACK + Native standalones, 3 variants per LV2, VST2, VST3, CLAP and AU.
This would easily get into 2Gb if each binary provided their own resources.

So on systems where installers are available (macOS and Windows) we use a common folder to store assets instead of being per binary.

You can already check this on the Windows releases. The exe installer is much smaller compared to the zip that works in a self-contained manner as you are describing.

PS: Regarding updating modules, PRs are welcome of course! Just take special note regarding licensing. Modules with VCV branding cannot be used as-is, they need their panels replaced, and we need to see about code license compatibilities too.
From a quick look at your changes, these modules are incompatible:
- CountModula
- SubmarineFree

**Comment by multivac61** [comment]
Thank you for the thorough answer. I understand your reservation regarding the binary sizes.
